server/services/v1/request_metadata.go

```diff
@@ -0,0 +1,101 @@
+package v1
+
+import (
+	"context"
+
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/server/transaction"
+)
+
+type RequestMetadataCtxKey struct {
+}
+
+type AccessToken struct {
+	Namespace string
+	Sub       string
+}
+
+type RequestMetadata struct {
+	accessToken AccessToken
+	namespace   metadata.Namespace
+}
+
+// NamespaceExtractor - extract the namespace from context
+type NamespaceExtractor interface {
+	Extract(ctx context.Context) (metadata.Namespace, error)
+}
+
+type AccessTokenNamespaceExtractor struct {
+	TenantManager      *metadata.TenantManager
+	Transactionmanager *transaction.Manager
+}
+
+func GetRequestMetadata(ctx context.Context) (*RequestMetadata, error) {
+	// read token
+	value := ctx.Value(RequestMetadataCtxKey{})
+	if value != nil {
+		if requestMetadata, ok := value.(*RequestMetadata); ok {
+			return requestMetadata, nil
+		}
+	}
+	return nil, api.Errorf(api.Code_NOT_FOUND, "request metadata not found")
+}
+
+func SetAccessToken(ctx context.Context, token AccessToken) context.Context {
+	requestMetadata, err := GetRequestMetadata(ctx)
+	var result = ctx
+	if err != nil && requestMetadata == nil {
+		requestMetadata = &RequestMetadata{}
+		result = context.WithValue(ctx, RequestMetadataCtxKey{}, requestMetadata)
+	}
+	requestMetadata.accessToken = token
+	return result
+}
+
+func SetNamespace(ctx context.Context, namespace metadata.Namespace) context.Context {
```

Comment:
so any Set on the context object shouldn't be here as that is done by middleware, only have getters in this file.

Reply:
updated.